File: multithreading/src/test/java/async/future/Service.java
package async.future;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class Service {
    private static Logger logger = LogManager.getLogger();

    public int getId() {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        logger.info("Return ID");
        return 1;
    }

    public int getBalanceById(int id) {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        logger.info("Return balance");
        return 1000 + id;
    }
}

File: kotlin/src/test/kotlin/coroutines/Service.kt
package coroutines

import mu.KotlinLogging

class Service {
    private val logger = KotlinLogging.logger { }
    fun getId(): Int {
        Thread.sleep(2000)
        logger.info("Return ID")
        return 1
    }

    fun getBalanceById(id: Int): Int {
        Thread.sleep(2000)
        logger.info("Return balance")
        return 1000 + id
    }
}
